src/rgw/driver/rados/rgw_user.cc

```diff
@@ -1555,7 +1555,7 @@ static void rename_swift_keys(const rgw_user& user,
   for ([[maybe_unused]] auto& [k, key] : modify_keys) {
     std::string id = user_id + ":" + key.subuser;
     key.id = id;
-    keys[id] = std::move(key);
+    keys[id] = std::move(key);  //NOLINT(bugprone-use-after-move)
```

Comment:
line 1554 seems like a real issue.
we move the `keys` parameter.
then we add new IDs into it at line 1558.
if the intent is to modify the key map into a new map, we should call `keys.clear()` after line 1554

Reply:
Hey @cbodley This issue was discussed in RGW standup can you please re-review this and provide an explanation on why this was marked as false-positive? Thank you.

Reply:
@yuvalif It definitely is an issue. Also, adding `clear` after moving `keys`, clang-tidy no longer generates the warning.

Reply:
Note to the reader: the issue discussed in the comments above was removed from this PR.